rust/src/delta.rs

```diff
@@ -1081,16 +1081,36 @@ impl DeltaTable {
         })
     }
 
-    /// Create a DeltaTable with version 0 given the provided MetaData and Protocol
+    /// Create a DeltaTable with version 0 given the provided MetaData, Protocol, and CommitInfo
     pub async fn create(
         &mut self,
         metadata: DeltaTableMetaData,
         protocol: action::Protocol,
+        commit_info: &mut Value,
     ) -> Result<(), DeltaTableError> {
         let meta = action::MetaData::try_from(metadata)?;
 
-        // TODO add commit info action
-        let actions = vec![Action::protocol(protocol), Action::metaData(meta)];
+        // delta-rs commit info will include the delta-rs version and timestamp as of now
+        let altered_commit_info = if commit_info.is_object() {
+            let temp = commit_info.as_object_mut().unwrap();
+            temp.insert(
+                "delta-rs".to_string(),
+                Value::String(crate_version().to_string()),
+            );
+            temp.insert(
+                "timestamp".to_string(),
+                Value::String(Utc::now().timestamp_millis().to_string()),
+            );
+            Some(Value::Object(temp.clone()))
+        } else {
+            None
+        };
+
+        let actions = vec![
+            Action::commitInfo(altered_commit_info.unwrap()),
+            Action::protocol(protocol),
+            Action::metaData(meta),
+        ];
```

Comment:
Here is a cleaner way to implement this logic without unwraps that could lead to hard crash at runtime:

```suggestion
        let mut enriched_commit_info = match commit_info {
            Value::Object(map) => Ok(map),
            _ => Err(<Return error here>),
        }?;
        enriched_commit_info.insert(
            "delta-rs".to_string(),
            Value::String(crate_version().to_string()),
        );
        enriched_commit_info.insert(
            "timestamp".to_string(),
            Value::String(Utc::now().timestamp_millis().to_string()),
        );

        let actions = vec![
            Action::commitInfo(Value::Object(enriched_commit_info)),
            Action::protocol(protocol),
            Action::metaData(meta),
        ];
```